Kibana CI complains about elastic-charts dependencies on xpack

**Describe the bug**
We have two different WIP PRs that integrates elastc-charts into Kibana that reports a red CI with the following error during the intake:

```
22:21:19 ERROR x-pack failed
22:21:19       ../node_modules/@elastic/charts/dist/lib/series/curves.d.ts:13:72 - error TS7016: Could not find a declaration file for module 'd3-shape'. '/var/lib/jenkins/workspace/elastic+kibana+pull-request/JOB/kibana-intake/node/immutable/kibana/node_modules/@elastic/charts/node_modules/d3-shape/dist/d3-shape.js' implicitly has an 'any' type.
22:21:19         If the 'd3-shape' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/d3-shape`
22:21:19 
22:21:19       13 export declare function getCurveFactory(curveType?: CurveType): import("d3-shape").CurveFactory;
22:21:19                                                                                 ~~~~~~~~~~
22:21:19 
22:21:19       ../node_modules/@elastic/charts/dist/utils/data/formatters.d.ts:1:26 - error TS7016: Could not find a declaration file for module 'luxon'. '/var/lib/jenkins/workspace/elastic+kibana+pull-request/JOB/kibana-intake/node/immutable/kibana/node_modules/luxon/build/node/luxon.js' implicitly has an 'any' type.
22:21:19         Try `npm install @types/luxon` if it exists or add a new declaration (.d.ts) file containing `declare module 'luxon';`
22:21:19 
22:21:19       1 import { Interval } from 'luxon';
22:21:19                                  ~~~~~~~
22:21:19 
22:21:19 
22:21:19       Found 2 errors.
```

Failing PRs: 
- https://github.com/elastic/kibana/pull/34313
- https://github.com/elastic/kibana/pull/33558

--- a/package.json
+++ b/package.json
@@ -49,7 +49,7 @@
     "@commitlint/cli": "^7.5.2",
     "@commitlint/config-conventional": "^7.5.0",
     "@elastic/datemath": "^5.0.2",
-    "@elastic/eui": "^8.0.0",
+    "@elastic/eui": "9.7.1",
     "@semantic-release/changelog": "^3.0.2",
     "@semantic-release/commit-analyzer": "^6.1.0",
     "@semantic-release/git": "^7.0.8",
@@ -67,12 +67,10 @@
     "@types/d3-collection": "^1.0.8",
     "@types/d3-random": "^1.1.2",
     "@types/d3-scale": "^2.1.1",
-    "@types/d3-shape": "^1.3.1",
     "@types/enzyme": "^3.9.0",
     "@types/enzyme-adapter-react-16": "^1.0.5",
     "@types/jest": "^24.0.9",
     "@types/lodash": "^4.14.121",
-    "@types/luxon": "^1.11.1",
     "@types/react": "^16.8.5",
     "@types/react-dom": "^16.8.2",
     "@types/storybook__addon-actions": "^3.4.2",
@@ -108,6 +106,8 @@
     "webpack": "^4.29.5"
   },
   "dependencies": {
+    "@types/d3-shape": "^1.3.1",
+    "@types/luxon": "^1.11.1",
     "classnames": "^2.2.6",
     "d3-array": "^2.0.3",
     "d3-collection": "^1.0.7",
@@ -123,14 +123,14 @@
     "prop-types": "^15.7.2",
     "react": "^16.8.3",
     "react-dom": "^16.8.3",
-    "react-konva": "^16.8.3",
+    "react-konva": "16.8.3",
     "react-spring": "^8.0.8",
     "resize-observer-polyfill": "^1.5.1"
   },
   "peerDependencies": {
     "@elastic/datemath": "^5.0.2",
-    "@elastic/eui": "^8.0.0",
-    "moment": "^2.24.0"
+    "@elastic/eui": "9.7.1",
+    "moment": "^2.20.1"
   },
   "config": {
     "commitizen": {

